components/MultiPanel/index.ts

```diff
@@ -1,12 +1,183 @@
 import './styles.css';
 
+/**
+ * A multi-panel view that the user can add any number of 'panels'. 
+ * 'a panel' consists of two elements. even index element begomes heading, 
+ * and odd index element becomes the expandable content.
+ */
 export default class MultiPanel extends HTMLElement {
+
   constructor () {
     super();
+
+    // Watch for children changes.
+    new MutationObserver(() => this._childrenChange())
+      .observe(this, { childList: true });
   }
 
   connectedCallback () {
+    this._childrenChange();
+
+    const children : Element[] = Array.from(this.children);
+    for (let i : number = 0; i < children.length; i++){
+      const child : Element = children[i]
+
+      if(i % 2 === 0){  
+        // even index = heading element
+        child.classList.add('panel-heading');
+        
+        // for A11y
+        child.id = `panel-heading-${i}`;
+        child.setAttribute('tabindex', '-1');
+        child.setAttribute('aria-controls', `panel-content-${i}`);
+      } else {
+        // odd index = content element
+        child.classList.add('panel-content');
+
+        // for A11y
+        child.id = `panel-content-${i-1}`;
+        child.setAttribute('aria-labelledby', `panel-heading-${i-1}`);
+      } 
+    }
+
+    // make the first heading focusable.
+    children[0].setAttribute('tabindex', '0');
```

Comment:
Is https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details a better primitive here in terms of accessibility? @surma did you ever look into this?

It might be that `<details>` buggers up too much when it comes to animation.